content/en/blog/wireshark.md

```diff
@@ -119,52 +118,108 @@ service PersonSearchService {
-
-To decode sample captures as gRPC message, you must decode the traffic on TCP
-port 50051 and 50052 as HTTP2:
+
+By selecting the **Load all files** option for the app's protocol directory you
+enable preloading of message definitions from the `addressbook.proto` file.
+
+### Setting port traffic type
+
+From the Wireshark [SampleCaptures page][], download the following app's sample gRPC capture file:
+[grpc_person_search_protobuf_with_image.pcapng][].
+
+<!--
+TODO: Why port 50052? I'm not seeing any traffic over port 50052 in the
+      screenshots. Is it actually used?
+-->
+The app's gRPC traffic was captured on ports 50051 and 50052. Before you can
+decode these messages, first register these ports as carrying HTTP2 traffic:
+
+<!--
+TODO: Clarify: how does the user access this configuration pane?
+TODO: Is the port 18127 relevant to this blog post, if not, can you remove it?
+TODO: Is the path "C:\Users\...decode_as_entries" actually part of the dialog? If not, can you remove it?
+-->
 ![wireshark_decode_as_dialog](/img/wireshark_decode_as_dialog.png)
 
-This is a screenshot of the gRPC service request message of
-[grpc_person_search_protobuf_with_image.pcapng](https://gitlab.com/wireshark/wireshark/-/wikis/uploads/f6fcdceb0248669c0b057bd15d45ab6f/grpc_person_search_protobuf_with_image.pcapng):
+### Decoding the search request message
+
+<!--
+TODO: What are the steps that the user needs to follow before the user can see
+      the decoded message as shown in the screenshot?
+-->
+Here's a screenshot of the sample's service request message:
+
+<!--
+TODO: Can you remove the commentary text (in blue) from the image. Such text is
+      best placed in the body of the post (it makes it easier to edit, and
+      eventually possibly translate into other languages :))
+
+TODO: If a users opens up the "GRPC_Message" section in that display, what will
+      be shown?
+-->
 ![wireshark_grpc_protobuf_search_request](/img/wireshark_grpc_protobuf_search_request.png)
-That shows searching for the Person objects of Jason and Lily based on their
-names.
+
+By examining the HTTP2 message header `path` field, you'll notice the URL to the
+app's service (`/tutorial.PersonSearchService`), followed by the name of the
+invoked RPC (`Search`).
+
+The `content-type`, which is set by the gRPC library, informs Wireshark that the
+HTTP2 message content is a gRPC message. By examining the decoded Protocol
+Buffers message of the sample gRPC request, you can see that the search request
+is for the names "Jason" and "Lily".
+
+### Decoding the server-streamed response
 
 Since the `Search` operation is defined as the server streaming RPC mode, the
 Person objects can be return back to client one after another:
+
+<!--
+
+TODO: The response is sent over port 51035. Does a user need to configure the
+      "decoding as" HTTP2 for this port?
+
+      It might also be worth mentioning to the reader how we know that 51035 is the response port.
```

Comment:
This is the TCP client port, that is random and need not "decoding as" because it will be different for each gRPC call. So, configuring "decoding as" on server port (50051) is enough.

Reply:
I've added a comment to this effect.